toolkit/tools/imagecustomizer/docs/configuration.md

```diff
@@ -0,0 +1,539 @@
+separate YAML (or JSON) file.
+
+The other YAML file schema is specified by [PackageList](#packagelist).
+
+Example:
+
+```yaml
+SystemConfig:
+  PackageListsRemove:
+  - lists/ssh.yaml
+```
+
+### PackagesInstall [string[]]
+
+Installs packages onto the image.
+
+Implemented by calling: `tdnf install`.
+
+Example:
+
+```yaml
+SystemConfig:
+  PackagesInstall:
+  - openssh-server
+```
+
+### PackageListsRemove [string[]]
+
+Same as [PackagesRemove](#packagesremove-string) but the packages are specified in a
+separate YAML (or JSON) file.
+
+The other YAML file schema is specified by [PackageList](#packagelist).
+
+Example:
+
+```yaml
+SystemConfig:
+  PackageListsRemove:
+  - lists/ssh.yaml
+```
+
+### PackagesRemove [string[]]
+
+Removes packages from the image.
+
+Implemented by calling: `tdnf remove`
+
+Example:
+
+```yaml
+SystemConfig:
+  PackagesRemove:
+  - openssh-server
+```
+
+### PackageListsUpdate [string[]]
+
+Same as [PackagesUpdate](#packagesupdate-string) but the packages are specified in a
+separate YAML (or JSON) file.
+
+The other YAML file schema is specified by [PackageList](#packagelist).
+
+Example:
+
+```yaml
+SystemConfig:
+  PackageListsRemove:
+  - lists/ssh.yaml
+```
+
+### PackagesUpdate [string[]]
+
+Updates packages on the system.
+
+Implemented by calling: `tdnf update`
+
+Example:
+
+```yaml
+SystemConfig:
+  PackagesUpdate:
+  - openssh-server
+```
+
+### AdditionalFiles [Map\<string, [FileConfig](#fileconfig)[]>]
+
+Copy files into the OS image.
+
+This property is a dictionary of source file paths to destination files.
+
+The destination files value can be one of:
+
+- The absolute path of a destination file.
+- A [FileConfig](#fileconfig) object.
+- A list containing a mixture of paths and [FileConfig](#fileconfig) objects.
+
+Example:
+
+```yaml
+SystemConfig:
+  AdditionalFiles:
+    # Single destination.
+    files/a.txt: /a.txt
+
+    # Single destinations with options.
+    files/b.txt:
+      Path: /b.txt
+      Permissions: "664"
+
+    # Multiple destinations.
+    files/c.txt:
+    - /c1.txt
+    - Path: /c2.txt
+      Permissions: "664"
+```
+
+### PostInstallScripts [[Script](#script)[]]
+
+Scripts to run against the image after the packages have been added and removed.
+
+These scripts are run under a chroot of the customized OS.
+
+Example:
+
+```yaml
+SystemConfig:
+  PostInstallScripts:
+  - Path: scripts/a.sh
+```
+
+### FinalizeImageScripts [[Script](#script)[]]
+
+Scripts to run against the image just before the image is finalized.
+
+These scripts are run under a chroot of the customized OS.
+
+Example:
+
+```yaml
+SystemConfig:
+  FinalizeImageScripts:
+  - Path: scripts/a.sh
+```
+
+### Users [[User](#user)]
+
+Used to add and/or update user accounts.
+
+Example:
+
+```yaml
+SystemConfig:
+  Users:
+  - Name: test
+```
+
+### Services [[Services](#services)]
+
+Options for configuring systemd services.
+
+```yaml
+SystemConfig:
+  Services:
+    Enable:
+    - sshd
+```
+
+## User
```

Comment:
duplicate to above ###Users

Reply:
looks like "Services" is the one out of context here

Reply:
`User` is a type. `Users` is a property on the `SystemConfig` type.

Reply:
I added `type` to the titles of the types to try to make this clearer.